atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -15,62 +15,99 @@
 
 @SuppressWarnings("checkstyle:FinalClass")
 public class SweepMetrics {
-    private final MetricsManager metricsManager = new MetricsManager();
-
-    private final TableSpecificHistogramMetric cellsSweptHistogram =
-            new TableSpecificHistogramMetric("cellTimestampPairsExamined");
-    private final TableSpecificHistogramMetric cellsDeletedHistogram =
-            new TableSpecificHistogramMetric("staleValuesDeleted");
+    private final TaggedMetricRegistry metricRegistry = new MetricsManager().getTaggedRegistry();
 
-    private final MeterMetric cellsSweptMeter = new MeterMetric("cellsSwept");
-    private final MeterMetric cellsDeletedMeter = new MeterMetric("cellsDeleted");
-    private final MeterMetric sweepErrorMeter = new MeterMetric("sweepError");
+    private final SweepMetric cellsSweptMetric = new SweepMetric("cellTimestampPairsExamined");
+    private final SweepMetric cellsDeletedMetric = new SweepMetric("staleValuesDeleted");
+    private final SweepMetric sweepTimeSweepingMetric = new SweepMetric("sweepTimeSweeping");
+    private final SweepMetric sweepTimeElapsedMetric = new SweepMetric("sweepTimeElapsedSinceStart");
+    private final SweepMetric sweepErrorMetric = new SweepMetric("sweepError");
 
-    private class TableSpecificHistogramMetric {
-        private final String name;
+    private class SweepMetric {
+        private final String histogram;
+        private final String meter;
+        private final String gauge;
 
-        TableSpecificHistogramMetric(String name) {
-            this.name = name;
+        SweepMetric(String name) {
+            this.histogram = name + "H";
+            this.meter = name + "M";
+            this.gauge = name + "G";
         }
 
-        void update(long value) {
-            metricsManager.getRegistry().histogram(aggregateMetric()).update(value);
+        void update(long value, Optional<TableReference> tableRef) {
+            metricRegistry.histogram(getTaggedMetric(histogram, tableRef)).update(value);
+            metricRegistry.meter(getTaggedMetric(meter, tableRef)).mark(value);
         }
 
-        private String aggregateMetric() {
-            return MetricRegistry.name(SweepMetrics.class, name);
+        void updateAggregate(long value) {
+            metricRegistry.histogram(getTaggedMetric(histogram, Optional.empty())).update(value);
+            ((Current) metricRegistry.gauge(getTaggedMetric(gauge, Optional.empty()), new Current())).setValue(value);
         }
+
     }
 
-    private class MeterMetric {
-        private final Meter meter;
+    @VisibleForTesting
+    static class Current implements Gauge<Long> {
+        private long value = 0;
+
+        @Override
+        public Long getValue() {
+            return value;
+        }
 
-        MeterMetric(String name) {
-            this.meter = metricsManager.registerOrGetMeter(SweepMetrics.class, null, name);
+        public void setValue(long newValue) {
+            value = newValue;
```

Comment:
This isn't thread safe because writing the long might not be an atomic operation.  You should make this either volatile or an AtomicLong.

Reply:
Well spotted!